modules/govuk_elasticsearch/manifests/backup.pp

```diff
@@ -0,0 +1,120 @@
+# == Class: govuk_elasticsearch::backup
+#
+# GOV.UK specific class to backup elasticsearch indexes.
+#
+# === Parameters
+#
+# [*aws_access_key_id*]
+#   AWS key to authenticate requests
+#
+# [*aws_secret_access_key*]
+#   AWS key to authenticate requests
+#
+# [*aws_region*]
+#   AWS region for the S3 bucket
+#
+# [*env_dir*]
+#   Defines directory for the environment
+#   variables
+#
+# [*s3_bucket*]
+#   Defines the AWS S3 bucket where the backups
+#   will be uploaded. It should be created by the
+#   user
+#
+# [*user*]
+#   Defines the system user that will be created
+#   to run the backups
+#
+# [*es_repos*]
+#   The repositories within elasticsearch
+#   where backups will be stored
+#
+# [*es_indices*]
+#   Elaticsearch indexes
+#
+# [*enabled*]
+#   Determines whether this job will be scheduled
+#
+# === Variables
+#
+# [*json_es_indices*]
+#   Formatted indexes to be backed up when script is invoked
+#
+class govuk_elasticsearch::backup(
+  $aws_access_key_id = undef,
+  $aws_secret_access_key = undef,
+  $aws_region = 'eu-west-1',
+  $s3_bucket = undef,
+  $env_dir = '/etc/es_s3backup',
+  $user = 'govuk-backup',
+  $es_repos = undef,
+  $enabled = false,
+  $es_indices = undef
+){
+
+  if $enabled {
```

Comment:
This should be removed and the gating should happen where this class is included.